### PR TITLE
Feature/alex go to hell

### PR DESCRIPTION
**Pull request by alexcmtk**

### CODE REVIEW

File: ICML-Submission/saliency.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+import numpy as np
+from matplotlib.patches import Ellipse
+import matplotlib.pyplot as plt
+import tensorflow.keras.backend as K
+from tensorflow.keras.layers import Input
+from tensorflow.keras.models import Model
+from tensorflow.keras.activations import softmax
+
+
+# change to vis saliency map
```
Comment (alexcmtk) [inline ICML-Submission/saliency.py:11]: Please rename this to what this is doing, turning a 3 channel image into a grayscale image. something like `convert_to_grayscale`. And snake_case.

File: ICML-Submission/saliency.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
+import tensorflow as tf
+import numpy as np
+from matplotlib.patches import Ellipse
+import matplotlib.pyplot as plt
+import tensorflow.keras.backend as K
+from tensorflow.keras.layers import Input
+from tensorflow.keras.models import Model
+from tensorflow.keras.activations import softmax
+
+
```
Comment (alexcmtk) [inline ICML-Submission/saliency.py:10]: Let's wrap these in a class. Something like `SaliencySuite`?

File: ICML-Submission/saliency.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    image_2d = np.sum(np.abs(image_3d), axis=2)
+    vmax = np.max(image_2d)
+    vmin = np.min(image_2d)
+    if vmax == vmin:
+        return np.zeros(image_2d.shape)
+    return (image_2d - vmin) / (vmax - vmin)
+
+def get_saliency_maps_and_fitted_ellipses(model, layers_index_list, no_of_nodes, image):
+    gradients = get_gradients_at_layers(model, layers_index_list, no_of_nodes, image)
+    reshaped_grads = np.reshape(gradients, (gradients.shape[0] * gradients.shape[1],) + gradients.shape[2:])
```
Comment (alexcmtk) [inline ICML-Submission/saliency.py:28]: Add a comment here explaining what this does. Then we don't need a test. So this turns (20, 30, 5, 2, 9) into (600, 5, 2, 9).
Can you comment what shape the gradients have and why this operation is necessary?

File: ICML-Submission/saliency.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+    saliency_map = np.full(reshaped_grads.shape[0:-1], np.nan)
+    
+    for i, gradient in enumerate(reshaped_grads):
+        saliency_map[i] = VisualizeImageGrayscale(gradient) 
+    
+    ellipses = generate_ellipses(saliency_map)
+    saliency_map = np.reshape(saliency_map, gradients.shape[0:-1])
```
Comment (alexcmtk) [inline ICML-Submission/saliency.py:36]: Why is it necessary to drop the last dimension?
